assignments/prototypes.js

```diff
@@ -32,7 +55,53 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
- 
+
+function Humanoid(humanoidAttrs) {
+  CharacterStats.call(this, humanoidAttrs);
+  this.team = humanoidAttrs.team;
+  this.weapons = humanoidAttrs.weapons;
+  this.language = humanoidAttrs.language;
+};
+
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+Humanoid.prototype.greet = function () {
+  return `${this.name} offers a greeting in ${this.language}.`;
+};
+
+
+// Stretch task: 
+
+  function Villian(villianAttrs){
+    Humanoid.call(this, villianAttrs);
+  };
+
+ Villian.prototype = Object.create(Humanoid.prototype);
+
+
+  Villian.prototype.attack = function () {
+    let villianAttack = Math.floor(Math.random() * 5) - this.healthPoints;
+    if(this.healthPoints <= 0){
+      console.log("Total Destruction")
+    }else{
+      console.log(`${this.name} has ${villianAttack} health points left! `);
+    }
+  };
+  
+  Hero.prototype = Object.create(Humanoid.prototype);
+
+  function Hero(heroAttrs){
+    Humanoid.call(this, heroAttrs);
+  };
+
+  Hero.prototype.attack = function () {
+    let heroAttack =  Math.floor(Math.random() * 5) - this.healthPoints;
+    if(this.healthPoints <= 0){
```

Comment:
It is somewhat of a convention to use a single white space between keywords *if*, *else*, ect. For example:
*if (condition) {
     <code block>
} else {
     <code block>
}

Reply:
Some say it makes it easier to read